src/entities.rs

```diff
@@ -78,7 +84,15 @@ impl<'de> serde::Deserialize<'de> for Entity {
         D: serde::Deserializer<'de>,
     {
         let bits = u64::deserialize(deserializer)?;
-        Ok(Entity::from_bits(bits))
+
+        if Entity::from_bits(bits).is_some() {
+            Ok(Entity::from_bits(bits).unwrap())
```

Comment:
`from_bits` should not be called twice. Use `match`, or `.ok_or_else(|| error_goes_here)?`.